common/component/redis/v9client.go

```diff
@@ -429,11 +443,53 @@ func newV9Client(s *Settings) RedisClient {
 			InsecureSkipVerify: s.EnableTLS,
 		}
 	}
-
+	client := v9.NewClient(options)
+	go refreshTokenRoutineForRedis(context.Background(), ClientFromV9Client(client), properties, v9logger, closeCh)
 	return v9Client{
-		client:       v9.NewClient(options),
+		client:       client,
 		readTimeout:  s.ReadTimeout,
 		writeTimeout: s.WriteTimeout,
 		dialTimeout:  s.DialTimeout,
+		closeCh:      closeCh,
+	}
+}
+
+func ClientFromV9Client(client v9.UniversalClient) RedisClient {
+	return v9Client{client: client}
+}
+
+func refreshTokenRoutineForRedis(ctx context.Context, redisClient RedisClient, meta map[string]string, logger logger.Logger, closeCh chan struct{}) {
+	ticker := time.NewTicker(time.Hour)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-closeCh:
+			return
+		case <-ticker.C:
+			env, err := azure.NewEnvironmentSettings(meta)
+			tokenCred, err := env.GetTokenCredential()
+			if err != nil {
+				logger.Debug("Failed to get Azure AD token credential:", err)
+				continue
+			}
+			at, err := tokenCred.GetToken(ctx, policy.TokenRequestOptions{
+				Scopes: []string{
+					env.Cloud.Services[azure.ServiceOSSRDBMS].Audience + "/.default",
+				},
+			})
+			if err != nil {
+				logger.Debug("Failed to get Azure AD token:", err)
+				continue
+			}
+
+			// Authenticate with Redis using the refreshed token
+			err = redisClient.(v9Client).client.Pipeline().Auth(ctx, at.Token).Err()
+			if err != nil {
+				logger.Debug("Failed to authenticate with Redis using refreshed Azure AD token:", err)
```

Comment:
Here too - `logger.Error` perhaps since the error isn't returned but just logged